File: ParadigmaComp/15Questoes/golang/questao5.go
package main

// Polimorfismo

import "fmt"

// Interface Animal
type Animal interface {
	Som() string
}

// Estrutura Cachorro
type Cachorro struct{}

func (c Cachorro) Som() string {
	return "O cachorro faz: Au Au"
}

// Estrutura Gato
type Gato struct{}

func (g Gato) Som() string {
	return "O gato faz: Miau"
}

func emitirSons(animais []Animal) {
	for _, animal := range animais {
		fmt.Println(animal.Som())
	}
}

func main() {
	// Exemplo de uso
	animais := []Animal{Cachorro{}, Gato{}}
	emitirSons(animais)
}

File: ParadigmaComp/15Questoes/python/questao5.py
# Polimorfismo
class Animal:
    def som(self):
        pass

class Cachorro(Animal):
    def som(self):
        return "O cachorro faz: Au Au"

class Gato(Animal):
    def som(self):
        return "O gato faz: Miau"

def emitir_sons(animais):
    for animal in animais:
        print(animal.som())

# Exemplo de uso
animais = [Cachorro(), Gato()]
emitir_sons(animais)
